### ISSUE TITLE
performance:r2() calculates the R2 in glmmTMB with beta distribution

### ISSUE THREAD

**Issue by cenabcd** [easystats/insight#349]
Hello contributor and all, 

when I use r2() to calcualte r2  with glmmTMB based on beta distribution , I cannot get the right answer. 
I will give my example here,

```r
Sy1 <- c(0.13236514,0.22750129,0.17552144,0.03125000,0.21581064,0.31045606,0.32317058,
          0.13786484,0.20363450,0.04667243,0.11466004,0.08553105,0.06910625,0.38306258,
          0.96875000,0.39076146)
Sy2 <- c(0.8379146,0.8839276,0.7977936,0.9687500,0.7088845,0.9071632,0.7015753,0.8983716,
         0.3673358,0.3851011,0.9039983,0.7838958,0.1089279,0.5363377,0.0312500,0.3579182)

Sx1 <- c(0.3721186,0.9342388,1.1827041,0.9062453,1.4028007,-0.2134339,-0.3026237, -0.4055091,
         0.9619032,1.0028019,-1.2453095,-1.4471977,-1.2453095,-1.4471977,-0.4175607,-0.1386708)

Sx2 <-c(0.3467040,-0.6168100,-1.0331197,0.2627958,-1.0300119,0.3591348,1.2634787,
         2.3511776,-0.9740731,-0.1505297,0.2255033,0.6885522,0.2255033,0.6201826,
        -1.2195823,-1.2289054)
Sx3 <- c(0.895777191,0.865860951,-1.321776649,0.500739605,1.126822462,0.004963455,
         1.287781382,1.361627154,-1.286865147,-0.182838527,-0.339620312,-1.474239471,
         -0.339620312,-1.474239471,0.155638668,0.219989023)
site <- c(rep(1,5),rep(2,3),rep(3,4),rep(4,4))
df <- data.frame(site,Sy1,Sy2,Sx1,Sx2,Sx3)

library(glmmTMB)
library(performance)
glmm1 <- glmmTMB(Sy1 ~ Sx1 + Sx2 + Sx3  + (1|site) ,family=beta_family(),data=df)
r2(glmm1)
#conditional R2: NA
#marginal R2:0.727

glmm2 <- glmmTMB(Sy2 ~ Sx1 + Sx2 + Sx3  + (1|site) ,family=beta_family(),data=df)
r2(glmm2)
#conditional R2: 1.019
#marginal R2:0.234
```
So what should I need to do? what do I need to pay attention to? Any hints would be greatly appreciated. Thank you already.

**Comment by strengejacke** [comment]
I don't think this is an issue due to possible misspecification of your model. Rather, this might be that the implementation for beta/beta-binomial distributions might not be reliable yet. I'll look into it. It's most likely due to the wrong computation of the random effects variances, that's why I moved this issue to the insight-package (`get_variance()`).

**Comment by strengejacke** [comment]
Should be fixed in #883 

``` r
Sy1 <- c(0.13236514,0.22750129,0.17552144,0.03125000,0.21581064,0.31045606,0.32317058,
          0.13786484,0.20363450,0.04667243,0.11466004,0.08553105,0.06910625,0.38306258,
          0.96875000,0.39076146)
Sy2 <- c(0.8379146,0.8839276,0.7977936,0.9687500,0.7088845,0.9071632,0.7015753,0.8983716,
         0.3673358,0.3851011,0.9039983,0.7838958,0.1089279,0.5363377,0.0312500,0.3579182)

Sx1 <- c(0.3721186,0.9342388,1.1827041,0.9062453,1.4028007,-0.2134339,-0.3026237, -0.4055091,
         0.9619032,1.0028019,-1.2453095,-1.4471977,-1.2453095,-1.4471977,-0.4175607,-0.1386708)

Sx2 <-c(0.3467040,-0.6168100,-1.0331197,0.2627958,-1.0300119,0.3591348,1.2634787,
         2.3511776,-0.9740731,-0.1505297,0.2255033,0.6885522,0.2255033,0.6201826,
        -1.2195823,-1.2289054)
Sx3 <- c(0.895777191,0.865860951,-1.321776649,0.500739605,1.126822462,0.004963455,
         1.287781382,1.361627154,-1.286865147,-0.182838527,-0.339620312,-1.474239471,
         -0.339620312,-1.474239471,0.155638668,0.219989023)
site <- c(rep(1,5),rep(2,3),rep(3,4),rep(4,4))
df <- data.frame(site,Sy1,Sy2,Sx1,Sx2,Sx3)

library(glmmTMB)
library(performance)
glmm1 <- glmmTMB(Sy1 ~ Sx1 + Sx2 + Sx3  + (1|site) ,family=beta_family(),data=df)
r2(glmm1, tolerance = 1e-10)
#> # R2 for Mixed Models
#> 
#>   Conditional R2: 0.319
#>      Marginal R2: 0.319
```

``` r

glmm2 <- glmmTMB(Sy2 ~ Sx1 + Sx2 + Sx3  + (1|site) ,family=beta_family(),data=df)
r2(glmm2, tolerance = 1e-10)
#> # R2 for Mixed Models
#> 
#>   Conditional R2: 0.835
#>      Marginal R2: 0.192
```

<sup>Created on 2024-06-12 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
